crates/oxc_ast/src/ast/comment.rs

```diff
@@ -97,86 +98,134 @@ pub struct Comment {
     #[estree(skip)]
     pub attached_to: u32,
 
-    /// Line or block comment
-    #[estree(rename = "type")]
-    pub kind: CommentKind,
-
-    /// Leading or trailing comment
+    /// Comment flags, which stores CommentAnnotation, CommentPosition, and CommentKind into one
     #[estree(skip)]
-    pub position: CommentPosition,
+    pub flags: CommentFlags,
+}
 
-    /// Whether this comment has a preceding newline.
-    /// Used to avoid becoming a trailing comment in codegen.
-    #[estree(skip)]
-    pub preceded_by_newline: bool,
+bitflags! {
+    /// Stores CommentAnnotation, CommentPosition, and CommentKind into one for memory efficiency
+    #[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+    pub struct CommentFlags: u16 {
+        /// The comment is a line comment.
+        const LINE = 1 << 0;
+        /// The comment is a block comment.
+        const BLOCK = 1 << 1;
+        /// The comment is a leading comment.
+        const LEADING = 1 << 2;
+        /// The comment is a trailing comment.
+        const TRAILING = 1 << 3;
```

Comment:
Aren't `LINE` and `BLOCK` mutually exclusive, and therefore only 1 bit needed for both? If `BLOCK` bit is set it's a block comment, if it's not set then it's a line comment.

Ditto `LEADING` / `TRAILING`.

Reply:
In the formatter there is some concept of `InlineBlock`, but generally speaking I think you are correct